logr_test.go

```diff
@@ -172,23 +172,26 @@ func TestError(t *testing.T) {
 }
 
 func TestV(t *testing.T) {
-	sink := &testLogSink{}
-	logger := New(sink)
-
-	if l := logger.V(0); l.level != 0 {
-		t.Errorf("expected level 0, got %d", l.level)
-	}
-	if l := logger.V(93); l.level != 93 {
-		t.Errorf("expected level 93, got %d", l.level)
-	}
-	if l := logger.V(70).V(6); l.level != 76 {
-		t.Errorf("expected level 76, got %d", l.level)
-	}
-	if l := logger.V(-1); l.level != 0 {
-		t.Errorf("expected level 0, got %d", l.level)
-	}
-	if l := logger.V(1).V(-1); l.level != 1 {
-		t.Errorf("expected level 1, got %d", l.level)
+	for name, logger := range map[string]Logger{
+		"testLogSink": New(&testLogSink{}),
+		"Discard":     Discard(),
+		"Zero":        {},
+	} {
+		t.Run(name, func(t *testing.T) {
+			adjust := func(level int) int {
+				if logger.GetSink() == nil {
+					// The Discard and the zero Logger short-cut the V call and don't
```

Comment:
If we have this `GetV` interface, maybe we should actually track the level, even when sink is nil.  Otherwise it's a weird corner case.

e.g.

```diff
diff --git a/logr.go b/logr.go
index 7dd08a6..8b58934 100644
--- a/logr.go
+++ b/logr.go
@@ -304,9 +304,6 @@ func (l Logger) Error(err error, msg string, keysAndValues ...any) {
 // level means a log message is less important.  Negative V-levels are treated
 // as 0.
 func (l Logger) V(level int) Logger {
-	if l.sink == nil {
-		return l
-	}
 	if level < 0 {
 		level = 0
 	}
```
 ?

Reply:
I tried and found that it breaks the API guarantee that any `Discard` instance is equal to any other, including [`Discard().V(2)`](https://github.com/go-logr/logr/blob/bde80be5ffb91839749c9baef150f6e00fa9e933/discard_test.go#L50-L53):

```
--- FAIL: TestComparable (0.00s)
    discard_test.go:52: any two discard Loggers must be equal
```

I agree that it would be nicer, but don't consider it worth breaking the API.